scripting/chat-processor.sp

```diff
@@ -159,18 +159,23 @@ public Action OnSayText2(UserMsg msg_id, BfRead msg, const int[] players, int pl
 
 	Handle hRecipients = CreateArray();
 	
+	// These while loops are probably very overkill but should theoretically eliminate any possible duplication.
+	int iIndex = -1;
+	
 	for (int i = 0; i < playersNum; i++)
 	{
-		if(FindValueInArray(hRecipients, players[i]) == -1) 
-		{
-			PushArrayCell(hRecipients, players[i]);
+		while ((iIndex = FindValueInArray(hRecipients, players[i]) != -1)) {
```

Comment:
This isn't needed here, since the hRecipients array is empty from the beginning, so your old solution is fine before executing the forward.